python/cugraph/traversal/sssp_wrapper.pyx

```diff
@@ -84,7 +84,7 @@ def sssp(input_graph, source):
     num_edges = input_graph.number_of_edges(directed_edges=True)
 
     # Step 5: Check if source index is valid
-    if not 0 <= source < num_verts:
+    if source is cudf.NA or 0 > source or num_verts <= source:
```

Comment:
It might be cleaner to raise an exception for `source is cudf.NA` in `sssp.py` after doing the renumbering conversion (if necessary).  I usually think of the `algorithm.py` file as dealing with things related to getting into the context for the calling C++ and the `algorithm_wrapper.pyx` file as dealing with the mechanical aspects of calling C++.